integration-tests/src/tests/network/ban_peers.rs

```diff
@@ -4,7 +4,7 @@ use std::time::Duration;
 /// Check we don't try to connect to a banned peer and we don't accept
 /// incoming connection from it.
 #[test]
-fn dont_connect_to_banned_peer() {
+fn dont_connect_to_banned_peer() -> anyhow::Result<()> {
```

Comment:
I'm not sure if you have seen how a stack trace of a thread managed by tokio looks like, especially if there are multiple runtimes running in parallel. With anyhow you can build up your stack trace to contain actually meaningful information. That's why I personally prefer Result passing and making the stacktrace contain only stuff that you explicitly add to it (and stuff added usually grows organically while debugging).

Reply:
Even with tokio, it' still useful to get a specific line where unwraps happens which you can open in the editor. But I think this part we actually run outside of tokio? Ie, I see why we might want to use anyhow in the guts of simulator, where we have many async things in flight, but this here is just standard sync code. 

No strong opinion here though.

Reply:
Ack, let me leave it as it is then for now. My last 2 days of debugging experience tell me that Result is worth a shot here. I'm open to rolling it back in case the next person touching it won't share my enthusiasm.